docs/chainlink-keepers/compatible-contracts.md

```diff
@@ -11,28 +11,57 @@ whatsnext:
+
+
+```solidity
+{% include samples/Keepers/KeepersCounter.sol %}
+```
+
+<div class="remix-callout">
+    <a href="https://remix.ethereum.org/#url=https://docs.chain.link/samples/Keepers/KeepersCounter.sol" >Open in Remix</a>
+    <a href="/docs/conceptual-overview/#what-is-remix" > What is Remix?</a>
+</div>
+
+
+The basic requirements are:
+1. Importing KeepersCompatible.sol. You can refer to the [Chainlink Github](https://github.com/smartcontractkit/chainlink) to find the latest version.
+2. Using the **`KeepersCompatibleInterface`** from the library to ensure your `checkUpkeep` and `performUpkeep`function definitions are what our Keeper Network expects. 
+3. A **`CheckUpkeep` function** that specifies the logic to be checked off-chain to see if `performUpkeep` should be executed. This is a simulation so it can use on-chain data.
+4. A **`performUpkeep` function** that will be executed on chain when `CheckUpkeep` returns `true`.
+
+
+To see more complex examples, please refer to our [utility contracts](../utility-contracts) page.
+
+We will now look at each function in more detail.
+
 
 ## Functions
 
 | Function Name                   | Description                                                          |
 | ------------------------------- | -------------------------------------------------------------------- |
-| [checkUpkeep](#checkupkeep-function)     | Checks if the contract requires work to be done.                     |
-| [performUpkeep](#performupkeep-function) | Performs the work on the contract, if instructed by `checkUpkeep()`. |
+| [checkUpkeep](#checkupkeep-function)     | Runs off-chain at every block to determine if on-chain function should be called.                     |
+| [performUpkeep](#performupkeep-function) | Contains the logic that should be executed on-chain once `checkUpkeep()` is true. |
 
 ### `checkUpkeep` function
 
-The Keeper node runs this method as an [`eth_call`](https://eth.wiki/json-rpc/API#eth_call) in order to determine if your contract requires some work to be done. If the off-chain simulation of your `checkUpkeep` confirms your predefined conditions are met, the Keeper will broadcast a transaction to the blockchain executing the `performUpkeep` method described below.
+This function contains the logic that determines when on-chain work should be done. It is off-chain and during at every block as an [`eth_call`](https://eth.wiki/json-rpc/API#eth_call). If this function simulation returns true, the Keeper will broadcast a transaction to the blockchain and execute the `performUpkeep` function.
```

Comment:
"It is off-chain and during at every block as an eth_call" doesn't feel right...maybe something like "It runs off-chain at every block as an eth_call:

Reply:
agree, updated